tidy3d/components/data/monitor_data.py

```diff
@@ -245,11 +242,17 @@ def _plane_grid_boundaries(self) -> Tuple[Coords1D, Coords1D]:
             # Last pixel missing compared to sim.discretize(self.monitor, extend=True).
             # This is because we cannot colocate the fields to the center of that pixel.
             # However, the monitor should be completely outside of this last pixel.
-            plane_bounds1 = tan_fields["E" + dim2].coords[dim1].values
-            plane_bounds2 = tan_fields["E" + dim1].coords[dim2].values
+            fields = self.symmetry_expanded_copy.field_components
+            plane_bounds1 = fields["E" + dim2].coords[dim1].values
+            plane_bounds2 = fields["E" + dim1].coords[dim2].values
 
         return plane_bounds1, plane_bounds2
 
+    @property
+    def _plane_grid_centers(self) -> Tuple[Coords1D, Coords1D]:
+        """For 2D monitor data, return the centers of the in-plane grid"""
+        return [(bs[1:] + bs[:-1]) / 2 for bs in self._plane_grid_boundaries]
```

Comment:
These "2D" methods all depend on `_tangetial_dims`, which does the validation and raises a proper error message. We could implement the validation on each one, but it seems excessive, specially because those are private methods, so we should catch any errors during development.